### PR TITLE
Token and bad code

### LINKED ISSUE #389: Token remover and auto codeblock response conflict
When a user posts code that's not formatted correctly with a discord.py token the bot deletes their message cause there's a token in it.  However it proceeds to repost their code **with** their token in a "how to format your code" response.

### PR DESCRIPTION
**Pull request by kraktus**
This PR fix #389 

Split the `token_remover` in two functions, one that checks that there is a token in the message (`is_token_in_message`), and the other takes the actions (deleting the message, etc.)
This new function `is_token_in_message` is reused in `cogs/bot.py` to know if the poorly formatted code contains a token. If it true, then the embed message is not displayed, since `token_remover.py` is already in charge.

### CODE REVIEW

File: bot/cogs/token_remover.py [hunk 1]
```diff
@@ -86,6 +75,23 @@ def mod_log(self) -> ModLog:
                 channel_id=Channels.mod_alerts,
             )
 
+    def is_token_in_message(self, msg: Message) -> bool:
```
Comment (jchristgit) [inline bot/cogs/token_remover.py]: This should probably be a classmethod, so you can call it from the codeblock cog without passing the class as the first argument, whilst still having access to the other methods.

File: bot/cogs/token_remover.py [hunk 2]
```diff
@@ -86,6 +75,24 @@ def mod_log(self) -> ModLog:
                 channel_id=Channels.mod_alerts,
             )
 
+    @classmethod
+    def is_token_in_message(self, msg: Message) -> bool:
```
Comment (jchristgit) [inline bot/cogs/token_remover.py]: I was expecting flake8 to nag here, but I guess there's no lint for it.

Via [PEP 8, the first argument to a classmethod should be named `cls`](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments)

Comment (kraktus) [reply bot/cogs/token_remover.py]: will be done, thanks

**Comment by kraktus** [comment]
@sco1 Hi, I am sorry but I am not sure about what you mean. Do you say the bot should also checks if a message contains a token every time a message is edited?

**Comment by sco1** [comment]
Yes

**Comment by lemonsaurus** [comment]
@kraktus any ETA on when you can add the requested changes from @sco1? We can't merge until that's done.

**Comment by kraktus** [comment]
Pretty busy right now, will do it as soon as I can.

**Comment by kraktus** [comment]
Not sure about using a `classmethod` or a `static method`.

**Comment by kraktus** [comment]
In case it is not clear, I changed my code to respond to the requested changes.

File: bot/cogs/bot.py [hunk 3]
```diff
@@ -237,7 +238,7 @@ def has_bad_ticks(self, msg: Message) -> bool:
             and len(msg.content.splitlines()) > 3
         )
 
-        if parse_codeblock:
+        if parse_codeblock and not TokenRemover.is_token_in_message(msg):  # no token in the msg
```
Comment (tagptroll1) [inline bot/cogs/bot.py]: Could this not be in parse_codeblock or a check that returns early before this?
```
if not TokenRemover.is_token_in_message(msg):
    return
...
```

**Comment by scragly** [comment]
A conflict was introduced here due to #681

**Pull request merged**